src/typings/product.tsx

```diff
@@ -1,14 +1,17 @@
 // export type ProductSizeTypes = "3xs" | "xxs" | "xs" | "s" | "m" | "l" | "xl" | "2xl" | "3xl";
 
+import SizeDialog from "../pages/MerchDetail/SizeDialog";
+
 export type ProductCategoryType = string;
 
 export type ProductType = {
   id: string;
   name: string;
   price: number;
+  stock: number[][]; // stock[colorway][size] = qty
+  sizes: string[];
+  colorways: string[];
   images?: string[];
-  sizes?: string[];
-  colorways?: string[];
   productCategory?: ProductCategoryType;
-  isAvailable?: boolean;
+  isAvailable: boolean;
```

Comment:
keeping `isAvailable` as a way to easily suspend/resume the sale of a product. not sure if this is necessary?

Reply:
I'd say we can keep it for now for potential "Coming Soon" items.

Reply:
In that case, should products that are not available be hidden from the merchandise list (landing page)?

Reply:
yup they should be hidden from all views @nicolelst

Reply:
may need to hide a product that has been discontinued

Reply:
Added badge to indicate products are unavailable/OOS from merchandise list page instead

Reply:
@RealDyllon got it, have hidden unavailable items from merchandise list. 
even if they access the merchandise details page by changing the product id in the url, there should be an `unavailable` badge and the purchasing buttons should be disabled.